front/components/agent_builder/instructions/AgentBuilderInstructionsContext.tsx

```diff
@@ -36,6 +38,8 @@ export function AgentBuilderInstructionsProvider({
       temperature: 0.7,
     });
 
+  const [instructions, setInstructions] = useState<string>("");
```

Comment:
nit: we really ought to improve that later, each keystroke is re-rendering the whole page